python/dolfinx/fem/forms.py

```diff
@@ -148,11 +149,29 @@ def _form(form):
                                         ]._cpp_object for i in range(ufcx_form.num_coefficients)]
         constants = [c._cpp_object for c in form.constants()]
 
-        # Subdomain markers (possibly None for some dimensions)
-        subdomains = {_cpp.fem.IntegralType.cell: subdomains.get("cell"),
-                      _cpp.fem.IntegralType.exterior_facet: subdomains.get("exterior_facet"),
-                      _cpp.fem.IntegralType.interior_facet: subdomains.get("interior_facet"),
-                      _cpp.fem.IntegralType.vertex: subdomains.get("vertex")}
+        # NOTE Could remove this and let the user convert mashtags by calling
+        # compute_integration_domains themselves
+        def get_integration_domains(integral_type, subdomains):
+            subdomain = subdomains.get(integral_type.name)
+            if isinstance(subdomain, dolfinx.mesh.MeshTagsMetaClass) \
```

Comment:
Alternatively, I could remove this function altogether and let the user call `compute_integration_domains` themselves if needed. This would be more explicit. What're your thoughts?